src/cirrus/build.py

```diff
@@ -100,16 +107,25 @@ def execute_build(opts):
             pypi_username=pypi_conf['username'],
             pypi_server=pypi_server
         )
-
-        cmd = (
-            '{0}/bin/pip install '
-            "-i {1} "
-            '-r {2}'
-            ).format(venv_path, pypi_url, reqs_name)
+        if opts.upgrade:
+            cmd = (
+                '{0}/bin/pip install --upgrade '
+                "-i {1} "
+                '-r {2}'
+                ).format(venv_path, pypi_url, reqs_name)
+        else:
+            cmd = (
+                '{0}/bin/pip install '
+                "-i {1} "
+                '-r {2}'
+                ).format(venv_path, pypi_url, reqs_name)
 
     else:
         # no pypi server
-        cmd = '{0}/bin/pip install -r {1}'.format(venv_path, reqs_name)
+        if opts.upgrade:
+            cmd = '{0}/bin/pip install --upgrade -r {1}'.format(venv_path, reqs_name)
+        else:
+            cmd = '{0}/bin/pip install -r {1}'.format(venv_path, reqs_name)
```

Comment:
In both these cases can't you just append --upgrade to the command, so you DRY for the rest of it? Disregard if that's not the case.